ee/api/test/test_organization.py

```diff
@@ -29,51 +29,96 @@ def test_delete_second_managed_organization(self):
         self.assertFalse(Organization.objects.filter(id=organization.id).exists())
         self.assertFalse(Team.objects.filter(id=team.id).exists())
 
-    def test_no_delete_last_organization(self):
+    def test_delete_last_organization(self):
         org_id = self.organization.id
         self.assertTrue(Organization.objects.filter(id=org_id).exists())
+
         response = self.client.delete(f"/api/organizations/{org_id}")
-        self.assertEqual(
-            response.data,
-            {
-                "attr": None,
-                "detail": f"Cannot remove organization since that would leave member {self.CONFIG_USER_EMAIL} organization-less, which is not supported yet.",
-                "code": "invalid_input",
-                "type": "validation_error",
-            },
-        )
-        self.assertEqual(response.status_code, 400)
-        self.assertTrue(Organization.objects.filter(id=org_id).exists())
 
-    def test_no_delete_organization_not_administrating(self):
-        organization, organization_membership, team = Organization.objects.bootstrap(self.user)
-        organization_membership = cast(OrganizationMembership, organization_membership)
-        organization_membership.level = OrganizationMembership.Level.MEMBER
-        organization_membership.save()
-        self.assertTrue(Organization.objects.filter(id=organization.id).exists())
-        self.assertTrue(Team.objects.filter(id=team.id).exists())
-        response = self.client.delete(f"/api/organizations/{organization.id}")
-        self.assertEqual(response.status_code, 403)
-        self.assertTrue(Organization.objects.filter(id=organization.id).exists())
-        self.assertTrue(Team.objects.filter(id=team.id).exists())
+        self.assertEqual(response.status_code, 204, "Did not successfully delete last organization on the instance")
+        self.assertFalse(Organization.objects.filter(id=org_id).exists())
+        self.assertFalse(Organization.objects.exists())
+
+        response_bis = self.client.delete(f"/api/organizations/{org_id}")
+
+        self.assertEqual(response_bis.status_code, 404, "Did not return a 404 on trying to delete a nonexistent org")
+
+    def test_no_delete_organization_not_owning(self):
```

Comment:
I would separate this into two tests, or remove the owner case altogether, makes the code confusing, and besides if the order of levels changes and `owner` is not last, this will lead to weird results.